diff --git a/.bazelrc b/.bazelrc
index 1d851908..081146c3 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -10,17 +10,13 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-common --enable_bzlmod
-common --incompatible_disallow_empty_glob
-common --incompatible_use_host_features
 common --lockfile_mode=off
 
 build --features=swift.use_explicit_swift_module_map
 build --host_features=swift.use_explicit_swift_module_map
 
 # Improved build performance
-build --host_swiftcopt=-whole-module-optimization
-build --swiftcopt=-whole-module-optimization
+build                       --@build_bazel_rules_swift//swift:copt=-whole-module-optimization
 
 # Keep in sync with Package.swift
 build --host_macos_minimum_os=10.15
diff --git a/.bazelversion b/.bazelversion
index a8907c02..acd405b1 100644
--- a/.bazelversion
+++ b/.bazelversion
@@ -1 +1 @@
-7.0.2
+8.6.0
diff --git a/BUILD.bazel b/BUILD.bazel
index 1f548b9b..b808d513 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -43,6 +43,7 @@ swift_syntax_library(
     name = "SwiftCompilerPluginMessageHandling",
     deps = [
         ":SwiftDiagnostics",
+        ":SwiftIfConfig",
         ":SwiftOperators",
         ":SwiftParser",
         ":SwiftSyntax",
@@ -291,6 +292,7 @@ swift_syntax_library(
     name = "SwiftSyntaxMacroExpansion",
     deps = [
         ":SwiftDiagnostics",
+        ":SwiftIfConfig",
         ":SwiftOperators",
         ":SwiftSyntax",
         ":SwiftSyntaxBuilder",
@@ -317,6 +319,7 @@ swift_syntax_library(
     name = "SwiftSyntaxMacros",
     deps = [
         ":SwiftDiagnostics",
+        ":SwiftIfConfig",
         ":SwiftParser",
         ":SwiftSyntax",
         ":SwiftSyntaxBuilder",
@@ -329,6 +332,7 @@ swift_syntax_library(
     deps = [
         ":SwiftDiagnostics",
         ":SwiftIDEUtils",
+        ":SwiftIfConfig",
         ":SwiftParser",
         ":SwiftSyntaxMacroExpansion",
         ":SwiftSyntaxMacros",
@@ -340,6 +344,7 @@ swift_syntax_library(
     name = "SwiftSyntaxMacrosTestSupport",
     testonly = True,
     deps = [
+        ":SwiftIfConfig",
         ":SwiftSyntax",
         ":SwiftSyntaxMacroExpansion",
         ":SwiftSyntaxMacros",
diff --git a/MODULE.bazel b/MODULE.bazel
index 25576534..036ddbbe 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -24,3 +24,4 @@ bazel_dep(
   max_compatibility_level = 3,
   repo_name = "build_bazel_rules_swift",
 )
+bazel_dep(name = "rules_cc", version = "0.2.17")
